### PR TITLE
Smaller additions to the hello_world example

### PR DESCRIPTION
**Pull request by barskern**
- Specify `Cargo.toml` for dependencies
- Include explanations of `tokio::{io,net}`
- Add section explaining how to run the code

### CODE REVIEW

File: content/docs/getting-started/hello-world.md [hunk 1]
```diff
@@ -38,6 +38,10 @@ use tokio::prelude::*;
 # fn main() {}
 ```
 
+Here we use Tokio's own [`io`] and [`net`] modules. These modules provide the same
+abstractions over networking and I/O-operations as the corresponding modules in `std`,
```
Comment (rylev) [inline content/docs/getting-started/hello-world.md]: nit: the `,` after `std` is not necessary

File: content/docs/getting-started/hello-world.md [hunk 2]
```diff
@@ -199,6 +203,26 @@ nothing more to do, so we just drop the socket, which closes it.
 
 You can find the full example [here][full-code]
 
+# Running the code
+
+To run the code we'll first have to setup a TCP server to listen for our connection. This
+can be done with [netcat].
```
Comment (rylev) [inline content/docs/getting-started/hello-world.md]: Is it redundant to explain `netcat` with one more sentence? "Netcat is a tool for quickly creating TCP and UDP sockets from the command line" - something like that?

File: content/docs/getting-started/hello-world.md (hunk 1 above)
Comment (barskern) [reply content/docs/getting-started/hello-world.md]: Agreed :+1:

File: content/docs/getting-started/hello-world.md (hunk 2 above)
Comment (barskern) [reply content/docs/getting-started/hello-world.md]: Yeah that does sound redundant. Rephrased it now to something similar to what you proposed.

**Comment by carllerche** [comment]
So sorry, PRs should be against `master`. I will try to fix this up.

**Comment by carllerche** [comment]
I have rebased this PR against master. The changes look good to me! ❤️

**Comment by carllerche** [comment]
I think I messed something up in the rebase...

**Comment by carllerche** [comment]
I think it is fixed now. @rylev feel free to merge once you verify that I did not mess things up.

**Comment by barskern** [comment]
I just moved the 'Fix rebase' into the corresponding commit so when it will be merge it'll look a bit cleaner 👍

**Comment by carllerche** [comment]
Thanks 👍

**Pull request merged**